File: 6-kyu/counting-duplicates/c++/solution.cpp
#include <algorithm>
#include <cctype>
#include <set>
#include <string>

size_t duplicateCount(const std::string& in); // helper for tests

size_t duplicateCount(const char* in)
{
    std::string str = in;
    std::transform(str.begin(), str.end(), str.begin(), ::tolower);
    std::set<char> char_set, counted_set;
    size_t ret = 0;
    for (const char &c : str) {
        if (char_set.count(c) == 0) char_set.insert(c);
        else if (counted_set.count(c) == 0) {
            ret++;
            counted_set.insert(c);
        }
    }
    return ret;
}

File: 6-kyu/counting-duplicates/python/solution.py
def duplicate_count(text):
    text = text.lower()
    char_set = []
    counted_set = []
    ret = 0
    for ch in text:
        if ch not in char_set:
            char_set.append(ch)
        elif ch not in counted_set:
            ret += 1
            counted_set.append(ch)
    return ret